optuna/notebooks/optuna_rapids.ipynb

```diff
@@ -0,0 +1,687 @@
+    "## Optuna\n",
+    "[Optuna](https://optuna.readthedocs.io/en/stable/) is a lightweight framework for automatic hyperparameter optimization. It provides a define-by-run API, which makes it easy to adapt to any already existing code that we have and enables high modularity along with the flexibility to construct hyperparameter spaces dynamically. By simply wrapping the objective function with Optuna, we can perform a parallel-distributed HPO search over a search space as we'll see in this notebook.\n",
+    "## Dask-Optuna\n",
+    "[Dask-Optuna](https://jrbourbeau.github.io/dask-optuna/#) is a library to help integrate Optuna's distributed optimizations to run trials on a Dask Cluster. It provides a `dask_optuna.DaskStorage` that wraps the Optuna storage class. This class helps extend in-memory computations to workers in the cluster.\n",
+    " \n",
+    "In this notebook, we'll use Airlines dataset (20M rows) to predict if a flight will be delayed or not. We'll explore how to use Optuna with RAPIDS in combination with Dask to run multi-GPU HPO experiments can yield results faster than CPU."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Run this cell to install optuna and mlflow\n",
+    "# !pip install optuna\n",
+    "# !pip install plotly\n",
+    "# !pip install -U kaleido\n",
+    "# !pip install dask_optuna\n",
+    "# !pip install 'bokeh<2.0.0'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "import time\n",
+    "from contextlib import contextmanager\n",
+    "\n",
+    "import cudf\n",
+    "import cuml\n",
+    "import dask_cudf\n",
+    "import numpy as np\n",
+    "import optuna\n",
+    "import pandas as pd\n",
+    "import sklearn\n",
+    "import os\n",
+    "import dask\n",
+    "import dask_optuna\n",
+    "\n",
+    "from cuml.dask.common import utils as dask_utils\n",
+    "from cuml.metrics import roc_auc_score, accuracy_score\n",
+    "from cuml.preprocessing.model_selection import train_test_split\n",
+    "from dask.distributed import Client, wait, performance_report\n",
+    "from joblib import parallel_backend, Parallel, delayed\n",
+    "from dask_cuda import LocalCUDACluster"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Helper function for timing blocks of code.\n",
+    "@contextmanager\n",
+    "def timed(name):\n",
+    "    t0 = time.time()\n",
+    "    yield\n",
+    "    t1 = time.time()\n",
+    "    print(\"..%-24s:  %8.4f\" % (name, t1 - t0))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up CUDA Cluster\n",
+    "\n",
+    "We start a local cluster and keep it ready for running distributed tasks with dask.\n",
```

Comment:
Maybe note that if you're running the Dask scheduler on a cluster, this will take advantage of multiple nodes automatically?